packages/interflex/src/__tests__/interflex.test.ts

```diff
@@ -0,0 +1,163 @@
+import Interflex from '../interflex';
+
+const enUS = {
+  localizedName: 'English',
+  strings: {
+    simple: 'Welcome!',
+    interpolate: 'Welcome, {name}.',
+    complex: 'Welcome! Please click <Link>here</Link>.',
+    complexInterpolate: 'Welcome, <foo>{name}</foo>.',
+    // NOTE: nested tags interpolation is still under discussion
+    // nestedComplexInterpolate:
+    //   'Welcome, <foo>{firstName} <bar>{lastName}</bar></foo>.',
+  },
+};
+
+describe('addLocaleData', () => {
+  let i18n = new Interflex<'en_US' | 'id_ID'>('en_US');
```

Comment:
This is kinda cool!

I'd consider possibly abstracting the constructor, by which I mean export a little helper `Interflex.create()` so that way the user doesn't know/care if we're using class or other approach.